src/JavaCodeGenerator.cs

```diff
@@ -4551,16 +4562,27 @@ private static void AddClientMethods(JavaType typeBlock, IEnumerable<ClientMetho
                             {
                                 GenericType restAPIMethodClientReturnType = (GenericType)ConvertToClientType(restAPIMethod.ReturnType);
                                 IType returnValueTypeArgumentClientType = restAPIMethodClientReturnType.TypeArguments.Single();
+                                IType mapperReturnType = clientMethod.ReturnValue.Type;
                                 if (restAPIMethodReturnBodyClientType != PrimitiveType.Void)
                                 {
                                     function.Text($".flatMapMaybe(");
-                                    function.Lambda(returnValueTypeArgumentClientType.ToString(), "res", "res.body() == null ? Maybe.empty() : Maybe.just(res.body())");
+                                    function.AnonymousClass($"new Function<{returnValueTypeArgumentClientType}, {mapperReturnType}>()", anonymousClassBlock =>
+                                    {
+                                        anonymousClassBlock.PublicMethod(
+                                            $"{mapperReturnType} apply({returnValueTypeArgumentClientType} res)",
+                                            method => method.Return($"res.body() == null ? Maybe.<{restAPIMethodReturnBodyClientType.AsNullable()}>empty() : Maybe.just(res.body())"));
+                                    });
                                     function.Line(");");
                                 }
                                 else if (isFluentDelete)
                                 {
                                     function.Text($".flatMapMaybe(");
-                                    function.Lambda(returnValueTypeArgumentClientType.ToString(), "res", "Maybe.empty()");
+                                    function.AnonymousClass($"new Function<{returnValueTypeArgumentClientType}, {GenericType.Maybe(ClassType.Void)}>()", anonymousClassBlock =>
+                                    {
+                                        anonymousClassBlock.PublicMethod(
+                                            $"{GenericType.Maybe(ClassType.Void)} apply({returnValueTypeArgumentClientType} res)",
+                                            method => method.Return("Maybe.empty()"));
+                                    });
```

Comment:
If you look at your usages of the AnonymousClass() function, you'll see that there is a pattern. They always take
```cs
$"new Function<{returnValueTypeArgumentClientType}, {return-type}>()"
```
as their first argument. I'd recommend inlining this into the body of the AnonymousClass function. so that AnonymousClass() takes the returnValueTypeArgumentClientType as its first parameter and return-type as its second argument.
You'll also notice that the interior call to PublicMethod() always looks the same too:
```cs
anonymousClassBlock.PublicMethod($"{return-type} apply({returnValueTypeARgumentClientType} res)", body-lambda);
```
You can abstract that away too by providing the `body-lambda` argument to AnonymousClass() so that your new invocation looks like this:
```cs
function.AnonymousClass(returnValueTypeArgumentClientType, return-type, body-lambda);
```
It's much smaller and much, much less prone to typos (did I remember to put the closing right arrow-bracket on the Function signature?).

Reply:
Renaming `AnonymousClass()` to `AnonymousFunction()` so that it's more reasonable to specialize for usability's sake. 👍